### PR TITLE
Split release process into major, minor and patch

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description  | Split release process into major, minor and patch. Add plenty missing informations in patch release process. Update some old references and remove some prestashop.com steps.
| Fixed ticket |

### CODE REVIEW

File: content/maintainers-guide/releasing-prestashop/minor/feature-freeze-process.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+### Register the new branch
+
+Following the creation of this new branch, update the following files to acknowledge the new branch:
+- [The Nightly builds][nightly-build]
+- [The Pull Request template][pr-template]
+
+## 3. Create the new version in Distribution API
+
+We don't know how to do it yet :D :D :D
```
Comment (Hlavtox) [inline content/maintainers-guide/releasing-prestashop/minor/feature-freeze-process.md:46]: 🤣

Comment (leemyongpakvn) [reply content/maintainers-guide/releasing-prestashop/minor/feature-freeze-process.md:46]: We will know how once we know who :walking:

Comment (matks) [reply content/maintainers-guide/releasing-prestashop/minor/feature-freeze-process.md:46]: Distribution API has been built on the following concept "Release X is acknowledge by the API when it is released" so it works OK for patch versions but for minor versions where you have a beta period, it fails. We need Distribution API to provide the information for 8.1.0 _before_ 8.1.0 is released not after.

No worries, Distribution API is young and it was expected we would miss some scenarii: it just means there is more to develop

**Comment by lartist** [comment]
In the process there is this command to run `php bin/console prestashop:linter:security-annotation`
When I try to run it, it seems like a argument is missing between "list" and "find-missing".
Maybe we should add it to the doc

**Comment by matks** [comment]
> In the process there is this command to run `php bin/console prestashop:linter:security-annotation` When I try to run it, it seems like a argument is missing between "list" and "find-missing". Maybe we should add it to the doc

Yes, it's because the command used to not require args and it changed. So the doc needs to be updated.

File: content/maintainers-guide/releasing-prestashop/minor/feature-freeze-process.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+{{% /notice %}}
+
+All actions below that request to submit a Pull Request must be done against this new branch.
+
+### Register the new branch
+
+Following the creation of this new branch, update the following files to acknowledge the new branch:
+- [The Nightly builds][nightly-build]
+- [The Pull Request template][pr-template]
+
```
Comment (lartist) [inline content/maintainers-guide/releasing-prestashop/minor/feature-freeze-process.md:43]:
```suggestion
- [Automatic tests](https://github.com/PrestaShop/ga.tests.ui.pr/pull/15/files)
```

File: content/maintainers-guide/releasing-prestashop/minor/_index.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    _To allow the Distribuion API to serve modules compatible with this new PrestaShop version._
+    
+   - **Update the version number in the Core.**
+   
+   - **Make sure the default translation catalogue has been updated and pushed to Crowdin.**  
+   _To make any new wordings translatable._
+
+   - **Create the branch for the theme.**  
+
+2. **[During stabilization][stabilization]**: (click to see the full step)
```
Comment (lartist) [inline content/maintainers-guide/releasing-prestashop/minor/_index.md:39]:
```suggestion
2. **[During stabilization][stabilization]**: (click to see the full step)
```

The link is not working

File: content/maintainers-guide/releasing-prestashop/major/_index.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+title: How to release a new major PrestaShop version
+menuTitle: Releasing a new major PS version
+chapter: true
```
Comment (eternoendless) [inline content/maintainers-guide/releasing-prestashop/major/_index.md:4]: This property doesn't exist in this template

**Review by eternoendless — COMMENTED** [review]
IMO splitting the process into major/minor/patch is not a good idea because 90% of the content is the same
It means every time we need to change something we need to do it in three places at once.

I think having a single list with "skips" is easier to maintain

File: content/maintainers-guide/releasing-prestashop/minor/_index.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+Minor versions go through the following [lifecycle][lifecycle].
+- Feature freeze
+- Public Beta period
+- Release Candidate
+- Final candidate
+
+
+## Process overview
+
+1. **[Perform feature freeze tasks][feature-freeze]**: (click to see the full step)
```
Comment (eternoendless) [inline content/maintainers-guide/releasing-prestashop/minor/_index.md:21]: This is a project step, not a release step. The release starts with the build.

**Comment by eternoendless** [comment]
I propose we document the "pre release" steps apart from the release steps themselves (which begin with the build). We could expand it to include ALL the steps in the release cycle (even those that precede feature freeze).

**Comment by matks** [comment]
> I think having a single list with "skips" is easier to maintain

It may be easier to maintain. But it is less usable and makes it hard for the release manager to know what exactly he should do.

Current release process is quite OK for patch versions but plenty steps for minor versions are not clear, must be done at another time and sometimes are not valid.

What is more important: whether it's easy to maintain or whether it's easy to use? Remember we are talking about a complex process that builds the very software that goes in 300,000 hands 😄 for me it's very obvious.

**Comment by eternoendless** [comment]
> plenty steps for minor versions are not clear

I think that the root problem is that it covers things that are outside scope of the release process. But if you look at it, there are practically no differences in the steps except in the "Preliminary tasks" part.

**Preliminary tasks**

1. "Create the new version in Addons Marketplace and update native module compatibility" - is only for 1.7.8.x releases and is on the way out
2. "Update version number in the code" - common to all types of releases
3. "Update translations" - only major and minor
4. "Lock the theme version" - common to all releases (BTW, how is this different from native module releases?)
5. "Manual verifications" - these should all be automated in CI

**Create a build**

All steps are common to all types of releases

**Release publicly**

All steps are common to all types of releases

**Final steps**

All steps are common to all types of releases

---

Based on the analysis above, I suggest only the "preliminary tasks" part to be split.

**Comment by matks** [comment]
@eternoendless All parts are different for patches, minor and major versions.

"Create a build" -> https://www.prestashop-project.org/maintainers-guide/releasing-prestashop/create-build/
> For major versions, we may build a Beta Version (example: prestashop_1.7.4.0-beta.1+build.1.zip).
> For patch versions, the beta and RC phase can be skipped (example: prestashop_1.7.4.1+build.1.zip)

"Release publicly" -> https://www.prestashop-project.org/maintainers-guide/releasing-prestashop/release-publicly/
> If you’re publishing a patch version:

It changes specifically because of how the theme is handled and because the description of the process depends if there is or not a freeze branch.

Minor versions use a freeze branch `8.1.x` while patch versions do not use a freeze branch.

"Final steps" -> https://www.prestashop-project.org/maintainers-guide/releasing-prestashop/final-steps/
> Check the PrestaShop localization packs are correct (only needed for major and minor releases):

It is _already_ a mess and complex to follow 😅 and this is going to become worse.

I understand that splitting the steps is going to increase maintainance: this is the price to pay to have a crystal clear release process that is as useful as possible for this sensitive action that is building a new release.

@lartist just followed the release process for 8.1.0 feature freeze: he can tell you that for someone who never did a release before (so no context and no ability to challenge what is written) it is very very hard to follow.

**Comment by kpodemski** [comment]
I agree with the idea of splitting instructions for each version type. If we want to keep it crystal-clear, then it should help us. We're not going to change the release process so often, so it shouldn't be a problem with maintenance.

One thing that please do make sure to add in the release process:

The release manager has to check all merged PRs, and see if they have a "Needs autoupgrade PR" label. If so, he needs to make sure that all PRs in the autoupgrade module have been created. If not, we need to create them. Before the final release and `autoupgrade` tests all pull requests need to be merged. We were struggling with this issue before 8.0, and it's the same problem with 8.1.
